src/validation.cpp

```diff
@@ -1277,6 +1277,10 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptMultipleTransactions(const std::
     // package feerate (total modified fees / total virtual size) to check this requirement.
     const auto m_total_vsize = std::accumulate(workspaces.cbegin(), workspaces.cend(), int64_t{0},
         [](int64_t sum, auto& ws) { return sum + ws.m_vsize; });
+    if (m_total_vsize > MAX_PACKAGE_SIZE * 1000) {
+        package_state.Invalid(PackageValidationResult::PCKG_POLICY, "package-too-large");
```

Comment:
in 9307e5c6fc1786e5fc734da1685e1cde52f4dd61

good catch, perhaps an error string like `package-exceeds-max-vsize` would help distinguish the two errors.

Reply:
Should they be distinguished?

Reply:
I think these are distinct rules (where "vsize" means sigop-adjusted):
(a) Each tx has a maximum vsize
(b) Ancestor and descendant packages within the mempool cannot exceed 101kvB vsize
(c) "Packages cannot exceed `MAX_PACKAGE_SIZE=101KvB` total size" ([doc](https://raw.githubusercontent.com/bitcoin/bitcoin/master/doc/policy/packages.md)).

The bug being fixed here is that we don't properly enforce (b) when we're validating packages.
I think your commit "Bugfix: Pass correct virtual size to CheckPackageLimits" can be followed by something like [this commit](https://github.com/bitcoin/bitcoin/pull/28450/commits/ad63201bd3c4b41395278f4b45934727471bacdf) instead of a new "package-too-large" check.

Imo the intent of (c) is to make sure we don't work on packages that are too large / potentially expensive to validate. For that, I think we should just change the description of (c) to "Packages cannot exceed `MAX_PACKAGE_SIZE=404kWu` total weight."